curation-migration-datasync/index.integration.ts

```diff
@@ -0,0 +1,305 @@
+      scheduledItemExternalId: 'random_scheduled_guid_2',
+      approvedItemExternalId: 'random_approved_guid_2',
+      lastUpdatedAt: timestamp1,
+    },
+    {
+      curatedRecId: 3,
+      scheduledSurfaceGuid: ScheduledSurfaceGuid.NEW_TAB_EN_US,
+      scheduledItemExternalId: 'random_scheduled_guid_3',
+      approvedItemExternalId: 'random_approved_guid_3',
+      lastUpdatedAt: timestamp1,
+    },
+    {
+      curatedRecId: 4,
+      scheduledSurfaceGuid: ScheduledSurfaceGuid.NEW_TAB_EN_GB,
+      scheduledItemExternalId: 'random_scheduled_guid_4',
+      approvedItemExternalId: 'random_approved_guid_4',
+      lastUpdatedAt: timestamp2,
+    },
+  ];
+
+  let db;
+  let testEventBody: AddScheduledItemPayload;
+  let testEvent;
+
+  beforeEach(async () => {
+    sinon.stub(SecretManager, 'getDbCredentials').resolves({
+      readHost: 'localhost',
+      readUsername: 'root',
+      readPassword: '',
+      writeHost: 'localhost',
+      writeUsername: 'root',
+      writePassword: '',
+      port: '3310',
+    });
+
+    db = await writeClient();
+
+    await truncateDynamoDb(dbClient);
+    await db(config.tables.curated_feed_prospects).truncate();
+    await db(config.tables.curated_feed_items).truncate();
+    await db(config.tables.curated_feed_queued_items).truncate();
+
+    testEventBody = {
+      eventType: EventDetailType.ADD_SCHEDULED_ITEM,
+      scheduledItemExternalId: 'random_scheduled_guid_1',
+      approvedItemExternalId: 'random_approved_guid_1',
+      url: 'https://stackoverflow.blog/',
+      title: 'Sync the new tool with legacy database',
+      excerpt: 'will be deprecated soon',
+      language: 'EN',
+      publisher: 'Pocket blog',
+      imageUrl: 'https://some-s3-url.com',
+      topic: 'SELF_IMPROVEMENT',
+      isSyndicated: false,
+      createdAt: 1648593897,
+      createdBy: 'ad|Mozilla-LDAP|sri',
+      updatedAt: 1648593897,
+      scheduledSurfaceGuid: 'NEW_TAB_EN_US',
+      scheduledDate: '2022-03-25',
+    };
+
+    testEvent = {
+      version: '0',
+      id: '8afc769f-bf1e-c87a-a0b6-1aa4c526831f',
+      'detail-type': 'add-scheduled-item',
+      source: 'curation-tools-datasync',
+      account: '124567',
+      time: '2022-03-29T22:55:16Z',
+      region: 'us-east-1',
+      resources: [],
+      detail: {
+        ...testEventBody,
+      },
+    };
+
+    nockParser(testEventBody);
+
+    //populating the database
+    const insertRecord = curatedItemRecords.map(async (item) => {
+      await insertCuratedItem(dbClient, item);
+    });
+    await Promise.all(insertRecord);
+
+    await db(config.tables.curated_feed_topics).truncate();
+    const inputTopicData = [
+      { topic_id: 1, name: 'Business', status: 'live' },
+      { topic_id: 2, name: 'Entertainment', status: 'live' },
+      { topic_id: 3, name: 'Health & Fitness', status: 'live' },
+      { topic_id: 4, name: 'Self Improvement', status: 'live' },
+    ].map((row) => {
```

Comment:
this map in unnecessary given the input array is the same as the resulting array